app/src/main/java/charlesli/com/personalvocabbuilder/controller/MainActivity.java

```diff
@@ -29,26 +30,35 @@
 import android.widget.TextView;
 import android.widget.Toast;
 
+import com.google.android.gms.appindexing.Action;
+import com.google.android.gms.appindexing.AppIndex;
+import com.google.android.gms.appindexing.Thing;
+import com.google.android.gms.common.api.GoogleApiClient;
+
 import charlesli.com.personalvocabbuilder.R;
 import charlesli.com.personalvocabbuilder.sqlDatabase.CategoryCursorAdapter;
 import charlesli.com.personalvocabbuilder.sqlDatabase.LanguageOptions;
 import charlesli.com.personalvocabbuilder.sqlDatabase.VocabDbContract;
 import charlesli.com.personalvocabbuilder.sqlDatabase.VocabDbHelper;
+import charlesli.com.personalvocabbuilder.ui.SettingsDialog;
 
 
 public class MainActivity extends AppCompatActivity {
 
     // Review Mode
     public static final int WORDTODEF = 0;
     public static final int DEFTOWORD = 1;
-    private final int DETECT_LANGUAGE = 0;
-    private final int ENGLISH = 19;
     private String reviewCategory = VocabDbContract.CATEGORY_NAME_MY_VOCAB;
     private int reviewMode = WORDTODEF;
     private int reviewNumOfWords = 0;
     private CategoryCursorAdapter mCategoryAdapter;
     private ListView mCategoryListView;
     private VocabDbHelper mDbHelper = VocabDbHelper.getDBHelper(MainActivity.this);
+    /**
```

Comment:
_NOTE:_ All this stuff about implementing the App Indexing API was something I've done accidentally with auto-correct using Android Studio (as mentioned, I am not used to that environment).

This is not the point of this pull-request, but you should probably read about anyway.